From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 8 Oct 2010 20:13:40 +0200
Subject: [PATCH] disable ordered compiling.

By default all modules are compiled in a predefined order. This breaks when
building not all modules. This patch disables ordered building. As a result
qmake generates the correct dependancies.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 examples/examples.pro |    4 ++++
 projects.pro          |    1 -
 src/src.pro           |    8 ++++++--
 3 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/examples/examples.pro b/examples/examples.pro
index f7d5306..67b4cf8 100644
--- a/examples/examples.pro
+++ b/examples/examples.pro
@@ -51,6 +51,10 @@ contains(DEFINES, QT_NO_GESTURES): SUBDIRS -= gestures
     SUBDIRS += multimedia
 }
 
+contains(QT_CONFIG, webkit) {
+    SUBDIRS += webkit
+}
+
 contains(QT_CONFIG, script): SUBDIRS += script
 
 contains(QT_CONFIG, phonon):!static: SUBDIRS += phonon
diff --git a/projects.pro b/projects.pro
index a2bc4b3..839670d 100644
--- a/projects.pro
+++ b/projects.pro
@@ -2,7 +2,6 @@
 # Main projectfile
 #####################################################################
 
-CONFIG += ordered
 TEMPLATE = subdirs
 
 cross_compile: CONFIG += nostrip
diff --git a/src/src.pro b/src/src.pro
index b9e48f4..4b1cf0d 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -100,12 +100,13 @@ src_declarative.target = sub-declarative
    src_qt3support.depends = src_gui src_xml src_network src_sql
    src_tools_idc.depends = src_corelib             # target defined in tools.pro
    src_tools_uic3.depends = src_qt3support src_xml # target defined in tools.pro
-   src_phonon.depends = src_gui
+   src_phonon.depends = src_gui src_network
    src_multimedia.depends = src_gui
    contains(QT_CONFIG, opengl):src_multimedia.depends += src_opengl
    src_activeqt.depends = src_tools_idc src_gui
    src_declarative.depends = src_gui src_script src_network
-   src_plugins.depends = src_gui src_sql src_svg
+   src_plugins.depends = src_gui src_sql
+   contains(QT_CONFIG, svg):src_plugins.depends += src_svg
    contains(QT_CONFIG, multimedia):src_plugins.depends += src_multimedia
    contains(QT_CONFIG, declarative):src_plugins.depends += src_declarative
    src_s60installs.depends = $$TOOLS_SUBDIRS $$SRC_SUBDIRS
@@ -114,6 +115,7 @@ src_declarative.target = sub-declarative
    contains(QT_CONFIG, webkit)  {
       src_webkit.depends = src_gui src_sql src_network
       contains(QT_CONFIG, xmlpatterns): src_webkit.depends += src_xmlpatterns
+      contains(QT_CONFIG, phonon): src_webkit.depends += src_phonon
       src_imports.depends += src_webkit
    }
    contains(QT_CONFIG, qt3support): src_plugins.depends += src_qt3support
@@ -133,6 +135,8 @@ src_declarative.target = sub-declarative
    contains(QT_CONFIG, svg) {
       src_declarative.depends += src_svg
    }
+   contains(QT_CONFIG, declarative):src_plugins.depends += src_declarative
+   contains(QT_CONFIG, phonon):src_plugins.depends += src_phonon
 }
 
 
